### PR TITLE
[DOC-2595] - CI iOS and macOS guide

### PR DESCRIPTION
**Pull request by aimurphy**
Preview: https://648a3055dbb52e048a7c1e52--harness-developer.netlify.app/tutorials/ci-pipelines/build/ios

### CODE REVIEW

**Comment by rustd** [comment]
@aimurphy thank you for writing this. Few comments below.

1. Remove this link [Kubernetes cluster pipeline tutorial](https://64839d038d9b5706341cbb02--harness-developer.netlify.app/tutorials/ci-pipelines/build/kubernetes-build-farm)
2. Self-hosted point to this tutorial (https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure)
3. Remove [Swift](https://www.swift.org/about/) is a popular iOS/macOS language. However, to make your app eligible for the Apple App Store, you must use the version of [Swift included with Xcode](https://www.swift.org/download/#:~:text=Using%20Downloads-,Apple%20Platforms,-Xcode%20includes%20a).
4. Add the yaml for self-hosted option
5. The "USE INTEL-BASED ARCHITECTURE" should be just a mention v/s highlighting it. M1 will be the default that majority of the customers will use
6. Install dependencies - Do we need this command?                       brew install xcode-archive-cache
7.  Remove the tip [Background steps]
8. Cache Intelligence (Hosted Cloud) Remove the tabs for Cache Intelligence, Save&Restore steps, Bitrise plugins. It should just have Cache Intelligence
9.  Rename "Run tests" section to "Build and run tests"
10. In Tests section, add the yaml examples for Hosted & Self-Hosted
11. Add a section for "Deploying to App Store"

**Comment by aimurphy** [comment]
@rustd Done except the following where I have questions:

- _Self-hosted point to this tutorial (https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure)_ - Previously you said to use the Anka guide.
- _Add the yaml for self-hosted option_ - Please specify where this is missing. In some cases the examples were the same so there was no need to differentiate with tabs.
- _Install dependencies - Do we need this command? brew install xcode-archive-cache:_ This was just an example. What dependencies they need is up to them.
- _In Tests section, add the yaml examples for Hosted & Self-Hosted_ - They are the same, AFAIK.
- _Add a section for "Deploying to App Store"_ - I need assistance writing this. I am not familiar with deployments. If it involves CD, then I need to pair with Michael or Roshni.

**Comment by aimurphy** [comment]
> @aimurphy Why do we have this for Cache Intelligence? ` - /Users/anka/Library/Developer/Xcode/DerivedData`

@rustd That was just something I put there because Idk what the correct cache path is.

**Comment by rustd** [comment]
- Have the yaml for both Hosted & Self-Hosted even though it could be the same. This will remove any confusion on if it was missing
- brew install fastlane is good enough to install dependencies.                       `brew install xcode-archive-cache `can be removed
- Deploying to AppStore doesn't need CD. Here's a pipeine. Can you please ping Eoin? https://app3.harness.io/ng/account/jW6eNyvET464tgD2xkxcXg/ci/orgs/default/projects/CI_Sanity/pipelines/osxdemo/executions/s9d2gXHXS-GIJVFJ2lKV9g/pipeline?storeType=INLINE

File: src/components/LandingPage/ContinuousIntegration.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default function CI() {
     //   </ul>
     <div className="container">
       <div className={styles.SectionName}>
-        <h3>Set up CI pipelines to automate builds, tests & publishing of artifacts</h3>
+        <h3>Set up CI pipelines to automate building, testing & publishing artifacts</h3>
```
Comment (brian-f-harness) [inline src/components/LandingPage/ContinuousIntegration.tsx:23]:
```suggestion
        <h3>Set up CI pipelines to automate building, testing, & publishing artifacts</h3>

```

File: tutorials/ci-pipelines/build/ci-ios.md [hunk 2]
```diff
@@ -0,0 +1,812 @@
...
+                  name: Test
+                  identifier: test
+                  spec:
+                    shell: Sh
+                    command: |-
+                      xcodebuild
+                      xcodebuild test -scheme SampleApp
+```
+
+If you want to [view test results in Harness](/docs/continuous-integration/use-ci/set-up-test-intelligence/viewing-tests/), make sure your test commands produce reports in JUnit XML format and that your steps include the `reports` specification. the following example uses [xcpretty](https://github.com/xcpretty/xcpretty) to produce reports in JUnit XML format.
```
Comment (brian-f-harness) [inline tutorials/ci-pipelines/build/ci-ios.md:275]:
```suggestion
If you want to [view test results in Harness](/docs/continuous-integration/use-ci/set-up-test-intelligence/viewing-tests/), make sure your test commands produce reports in JUnit XML format and that your steps include the `reports` specification. The following example uses [xcpretty](https://github.com/xcpretty/xcpretty) to produce reports in JUnit XML format.
```

File: tutorials/ci-pipelines/build/ci-ios.md [hunk 3]
```diff
@@ -0,0 +1,812 @@
...
```
Comment (brian-f-harness) [inline tutorials/ci-pipelines/build/ci-ios.md:320]:
```suggestion
If you want to [view test results in Harness](/docs/continuous-integration/use-ci/set-up-test-intelligence/viewing-tests/), make sure your test commands produce reports in JUnit XML format and that your steps include the `reports` specification. The following example uses [xcpretty](https://github.com/xcpretty/xcpretty) to produce reports in JUnit XML format.
```

**Review by brian-f-harness — DISMISSED** [review]
lgtm

**Pull request merged**